### PR TITLE
refactor CustomClaimConverter

### LINKED ISSUE #3973: Refactor CustomClaimConverter
Some pain points: 

* The rest access is done without any timeout (this can quickly be a real problem);
* The users cache is not really easy to understand, using a memoizer (from `java-memoizers`) can be a good idea.

### PR DESCRIPTION
**Pull request by renanfranca**
Fix #3973

### CODE REVIEW

**Comment by renanfranca** [comment]
I implemented the first pain point:
> The rest access is done without any timeout (this can quickly be a real problem);

I am going to implement the second pain point:
> The users cache is not really easy to understand, using a memoizer (from java-memoizers) can be a good idea.

File: src/main/resources/generator/server/springboot/mvc/security/oauth2/core/main/infrastructure/primary/CustomClaimConverter.java.mustache [hunk 1]
```diff
@@ -82,14 +83,18 @@ class CustomClaimConverter implements Converter<Map<String, Object>, Map<String,
...
+      try {
+        ResponseEntity<ObjectNode> userInfo = restTemplate.exchange(
+          registration.getProviderDetails().getUserInfoEndpoint().getUri(),
+          HttpMethod.GET,
+          new HttpEntity<String>(headers),
+          ObjectNode.class
+        );
+
+        return userInfo.getBody();
+      } catch (ResourceAccessException ex) {
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/mvc/security/oauth2/core/main/infrastructure/primary/CustomClaimConverter.java.mustache:95]: It's better to rethrow the exception in this case I think

Comment (renanfranca) [reply src/main/resources/generator/server/springboot/mvc/security/oauth2/core/main/infrastructure/primary/CustomClaimConverter.java.mustache]: I agree! I pushed a new version. Thank you @DamnClin

**Comment by renanfranca** [comment]
I implemented the second pain point:
> The users cache is not really easy to understand, using a memoizer (from java-memoizers) can be a good idea.

**Comment by renanfranca** [comment]
The integration test failed because I am using the Memoizers module at OAuth2Module. I've already put 
Memoizers module as a dependency of OAuth2Module. I don't know how to change the integration test to add MemoizersModule when generating the oauth2app. I face the same problem when I run the integration test locally: `./tests-ci/generate.sh oauth2app`

Thank you in advance 😃

**Comment by DamnClin** [comment]
Add it here https://github.com/jhipster/jhipster-lite/blob/main/tests-ci/generate.sh

**Comment by renanfranca** [comment]
Thanks, @DamnClin !!! All checks have passed 😄

**Comment by renanfranca** [comment]
Bounty Claimed [https://opencollective.com/generator-jhipster/expenses/121936](https://opencollective.com/generator-jhipster/expenses/121936) 😃

Thanks, @pascalgrimaud!

**Comment by pascalgrimaud** [comment]
@renanfranca : approved, thanks for your work

**Pull request merged**